### PR TITLE
Some improvements to metadata feature

### PR DESCRIPTION
**Pull request by DaughterOfMars**

### CODE REVIEW

File: sdk/src/types/block/output/feature/metadata.rs [hunk 1]
```diff
@@ -35,72 +35,29 @@ type MetadataBTreeMap =
...
             }
         }
     }
     Ok(())
 }
 
-fn verify_length_packable<const VERIFY: bool>(map: &MetadataBTreeMapPrefix) -> Result<(), Error> {
+fn verify_byte_length_packable<const VERIFY: bool>(len: usize) -> Result<(), Error> {
     if VERIFY {
         // +1 for the feature type
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/feature/metadata.rs]: What happened to this?

File: sdk/src/types/block/output/feature/metadata.rs [hunk 2]
```diff
@@ -114,73 +71,163 @@ fn verify_length_packable<const VERIFY: bool>(map: &MetadataBTreeMapPrefix) -> R
...
     type Error = Error;
 
-    fn try_from(data: BTreeMap<Vec<u8>, Vec<u8>>) -> Result<Self, Error> {
-        metadata_feature_from_iter(data)
+    fn try_from(data: Vec<(String, Vec<u8>)>) -> Result<Self, Error> {
+        let mut builder = Self::build();
+        for (k, v) in data {
+            builder.insert(k, v);
+        }
+        builder.finish()
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/feature/metadata.rs]: Worth it to move this into a function? It's three times the same thing

File: sdk/src/types/block/output/feature/metadata.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/types/block/output/feature/metadata.rs]: I found that I could not use this in the unpack impl unless I changed the fn

File: sdk/src/types/block/output/feature/metadata.rs (hunk 2 above)
Comment (DaughterOfMars) [reply sdk/src/types/block/output/feature/metadata.rs]: could be, though I don't care too much

File: sdk/src/types/block/output/feature/metadata.rs (hunk 1 above)
Comment (Thoralf-M) [reply sdk/src/types/block/output/feature/metadata.rs]: Yes but the +1 is gone?

Comment (DaughterOfMars) [reply sdk/src/types/block/output/feature/metadata.rs]: because I'm packing the whole feature

Comment (Thoralf-M) [reply sdk/src/types/block/output/feature/metadata.rs]: But the type is in the feature enum, not in the feature itself

Comment (DaughterOfMars) [reply sdk/src/types/block/output/feature/metadata.rs]: ah, whoops 😆

Comment (Thoralf-M) [reply sdk/src/types/block/output/feature/metadata.rs]: Actually checked the TIP again and the limit is without the type, so just removing the comment

**Pull request merged**